src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StackAlloc.cs

```diff
@@ -46,7 +46,8 @@ public override BoundNode VisitStackAllocArrayCreation(BoundStackAllocArrayCreat
                 var locals = ArrayBuilder<LocalSymbol>.GetInstance();
                 var countTemp = CaptureExpressionInTempIfNeeded(rewrittenCount, sideEffects, locals, SynthesizedLocalKind.Spill);
                 var stackSize = RewriteStackAllocCountToSize(countTemp, elementType);
-                stackAllocNode = new BoundConvertedStackAllocExpression(stackAllocNode.Syntax, elementType, stackSize, initializerOpt, spanType);
+                stackAllocNode = new BoundConvertedStackAllocExpression(
+                    stackAllocNode.Syntax, elementType, stackSize, initializerOpt, _compilation.CreatePointerTypeSymbol(elementType));
```

Comment:
>_compilation.CreatePointerTypeSymbol(elementType) [](start = 83, length = 49)

I didn't understand why this part of the change is necessary or good.

Reply:
This expression is being passed as an argument to the constructor of `Span<T>`.  The argument is a use of the `stackalloc` instruction, which produces a pointer.  But the compiler was recording the type of the argument expression as `Span<T>` instead of a pointer type.  This change corrects that bug.  There were no symptoms of the bug, but it makes the compiler fragile to have nodes lying about their types.

---
In reply to: [285221496](https://github.com/dotnet/roslyn/pull/35780#discussion_r285221496) [](ancestors = 285221496)